docs/device-api.md

```diff
@@ -0,0 +1,106 @@
+# Device-API Functionality
+
+The Device-API following functionality matrix indicates which devices are
+required for an API call to be usable.
+
+**O** (Optional): The device (column) **is not required** for a Firecracker
+microVM to run. If the device (column) is omitted from a kernel build the API
+element (row) is **TBA**.
+
+**R** (Required): The device (column) **is required** for a Firecracker microVM
+to run. If the device (column) is omitted from a kernel build the API
+element (row) response is **TBA**.
+
+## API Endpoints
```

Comment:
Can we mention that the API endpoints are described in the [firecracker swagger file](https://github.com/firecracker-microvm/firecracker/blob/master/src/api_server/swagger/firecracker.yaml)?

Reply:
Sure, but it seems extraneous: Or at least I generally don't care how an API is implemented, and I don't want to think I have to understand swaggers peculiarities to know how to use this API.